src/maple_pool/depository.ts

```diff
@@ -0,0 +1,225 @@
+
+export class MaplePoolDepository {
+  public static SYRUP_PROGRAM_ID = new PublicKey(
+    '5D9yi4BKrxF8h65NkVE1raCCWFKUs5ngub2ECxhvfaZe'
+  );
+
+  public static POOL_CREDORA_USDC = {
+    maplePool: new PublicKey('TamdAwg85s9aZ6mwSeAHoczzAV53rFokL5FVKzaF1Tb'),
+    maplePoolLocker: new PublicKey(
+      '92oAd9cm4rV4K4Xx9HPRMoFn7GwMaKsjNSPe7QVxywcy'
+    ),
+    mapleGlobals: new PublicKey('DtnAPKSHwJaYbFdjYibNcjxihVd6pK1agpT86N5tMVPX'),
+    mapleSharesMint: new PublicKey(
+      'CesxqgX4BvYudTNU45PArqTgefrRFhE1CwR7ECTDshfY'
+    ),
+  };
+
+  public constructor(
+    public readonly pda: PublicKey,
+    public readonly collateralMint: PublicKey,
+    public readonly collateralDecimals: number,
+    public readonly collateralSymbol: string,
+    public readonly depositoryCollateral: PublicKey,
+    public readonly maplePool: PublicKey,
+    public readonly maplePoolLocker: PublicKey,
+    public readonly mapleGlobals: PublicKey,
+    public readonly mapleLender: PublicKey,
+    public readonly mapleSharesMint: PublicKey,
+    public readonly mapleLockedShares: PublicKey,
+    public readonly mapleLenderShares: PublicKey,
+    public readonly syrupProgramId: PublicKey
+  ) {}
+
+  public static async initialize({
+    connection,
+    uxdProgramId,
+    syrupProgramId,
+    collateralMint,
+    collateralSymbol,
+    maplePool,
+    maplePoolLocker,
+    mapleGlobals,
+    mapleSharesMint,
+  }: {
+    connection: Connection;
+    uxdProgramId: PublicKey;
+    syrupProgramId: PublicKey;
+    collateralMint: PublicKey;
+    collateralSymbol: string;
+    maplePool: PublicKey;
+    maplePoolLocker: PublicKey;
+    mapleGlobals: PublicKey;
+    mapleSharesMint: PublicKey;
+    cluster: Cluster;
+  }): Promise<MaplePoolDepository> {
+    // The depository is PDA from the pool and the collateral
+    const [depository] = PublicKey.findProgramAddressSync(
+      [
+        Buffer.from(MAPLE_POOL_DEPOSITORY_NAMESPACE),
+        maplePool.toBuffer(),
+        collateralMint.toBuffer(),
+      ],
+      uxdProgramId
+    );
+
+    // Read collateral decimals
+    const collateralToken = new Token(
+      connection,
+      collateralMint,
+      TOKEN_PROGRAM_ID,
+      null as unknown as Signer
+    );
+    const collateralInfo = await collateralToken.getMintInfo();
+    if (!collateralInfo) {
+      throw new Error('Cannot find the collateral mint');
+    }
+    const collateralDecimals = collateralInfo.decimals;
+
+    const depositoryCollateral = await this.findDepositoryCollateralAddress(
+      depository,
+      uxdProgramId
+    );
+
+    const mapleLender = await this.findLenderAddress(
+      maplePool,
+      depository,
+      syrupProgramId
+    );
+
+    const mapleLockedShares = await this.findLockedSharesAddress(
+      mapleLender,
+      syrupProgramId
+    );
+    const mapleLenderShares = await this.findLenderSharesAddress(
+      depository,
+      mapleSharesMint
+    );
```

Comment:
May be interesting to use `Promise.all` to boost the speed a bit

```suggestion
    const [
        collateralInfo,
        depositoryCollateral,
        mapleLender,
        mapleLenderShares,
    ] = await Promise.all([
        collateralToken.getMintInfo(),
        
        this.findDepositoryCollateralAddress(
            depository,
            uxdProgramId
        ),

        this.findLenderAddress(
            maplePool,
            depository,
            syrupProgramId
        ),

        this.findLenderSharesAddress(
            depository,
            mapleSharesMint
        ),
    ]);

    if (!collateralInfo) {
      throw new Error('Cannot find the collateral mint');
    }
    const {
        decimals: collateralDecimals,
    } = collateralInfo;
    
    const mapleLockedShares = await this.findLockedSharesAddress(
      mapleLender,
      syrupProgramId
    );
```
```